borg/helpers.py

```diff
@@ -307,52 +307,60 @@ def parse_timestamp(timestamp):
         return datetime.strptime(timestamp, '%Y-%m-%dT%H:%M:%S').replace(tzinfo=timezone.utc)
 
 
-def load_excludes(fh):
-    """Load and parse exclude patterns from file object. Lines empty or starting with '#' after stripping whitespace on
-    both line ends are ignored.
-    """
-    patterns = (line for line in (i.strip() for i in fh) if not line.startswith('#'))
-    return [parse_exclude_pattern(pattern) for pattern in patterns if pattern]
+def parse_add_pattern(patternstr, roots, patterns):
+    """Parse a pattern string and add it to roots or patterns depending on the pattern type."""
+    pattern = parse_inclexcl_pattern(patternstr)
+    if pattern.ptype is RootPath:
+        roots.append(pattern.pattern)
+    else:
+        patterns.append(pattern)
 
 
-def load_patterns(fh):
-    """Load and parse include/exclude/root patterns from file object.
-    Lines empty or starting with '#' after stripping whitespace on both line ends are ignored.
-    """
-    patternlines = (line for line in (i.strip() for i in fh) if not line.startswith('#'))
-    roots = []
-    inclexcl_patterns = []
-    for patternline in patternlines:
-        pattern = parse_inclexcl_pattern(patternline)
-        if pattern.ptype is RootPath:
-            roots.append(pattern.pattern)
-        else:
-            inclexcl_patterns.append(pattern)
-    return roots, inclexcl_patterns
+def pattern_file_iter(fileobj):
+    for line in fileobj:
+        line = line.strip()
+        if not line or line.startswith('#'):
+            continue
+        yield line
```

Comment:
iirc i have added similar code to master branch, please just backport that utility function, so we are in sync.